### PR TITLE
[IMP] website: redesign `s_product_list` snippet

### PR DESCRIPTION
**Pull request by stefanorigano**
----
PR/Task handled by @anso-odoo 

----

The goal of this commit is to improve the visual design of the default `s_product_list` snippet.

task-4048196

Requires: https://github.com/odoo/design-themes/pull/848

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by anso-odoo** [comment]
@Brieuc-brd this one is 🟢

File: addons/website/views/snippets/s_product_list.xml [hunk 1]
```diff
@@ -3,62 +3,94 @@
 
 <template name="Items" id="s_product_list">
     <t t-set="url" value="url or '#'"/>
-    <section class="s_product_list">
-        <div class="container-fluid">
+    <section class="s_product_list pt64 pb64" data-vcss="001">
+        <div class="container">
+            <h3 class="mb-4">Our finest selection</h3>
```
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_product_list.xml:8]: Doing that will prevent the customization made in the WB, you should remove it :) 
Capture d’écran 2024-08-01 à 16 39 16

File: addons/website/views/snippets/s_product_list.xml [hunk 2]
```diff
@@ -3,62 +3,94 @@
...
                 <div class="col-lg-2 col-md-4 col-6">
-                    <a t-att-href="url">
-                        <img src="/web/image/website.s_product_list_default_image_1" alt="" class="img img-fluid"/>
+                    <a t-att-href="url" class="text-decoration-none">
+                        <div class="s_card o_card_img_top card o_cc o_cc1" data-name="Card" data-vcss="001">
+                            <figure class="o_card_img_wrapper ratio ratio-4x3 mb-0">
+                                <img src="/web/image/website.s_product_list_default_image_1" alt="" class="o_card_img card-img-top"/>
+                            </figure>
+                            <div class="card-body text-center">
+                                <h6 class="card-title m-0">Elegant</h6>
```
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_product_list.xml:17]: Same as my previous comment, it prevents the margin customization available in the WB.
```suggestion
                                <h6 class="card-title">Elegant</h6>
```
Same for the other cards

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
First part @anso-odoo  👍

**Comment by anso-odoo** [comment]
Thanks, fixed; can you check the rest ?

File: addons/website/views/snippets/s_product_list.xml [hunk 3]
```diff
@@ -3,62 +3,93 @@
 
 <template name="Items" id="s_product_list">
     <t t-set="url" value="url or '#'"/>
-    <section class="s_product_list">
-        <div class="container-fluid">
+    <section class="s_product_list pt64 pb64" data-vcss="001">
+        <div class="container">
+            <h3>Our finest selection</h3>
```
Comment (Brieuc-brd) [inline addons/website/views/snippets/s_product_list.xml:8]: I'd place the title in a column. This way, the user could easily customize the spacing and alignment.

```suggestion
<div class="col-lg-12">
   <h3>Our finest selection</h3>
</div>
```

**Review by Brieuc-brd — CHANGES_REQUESTED** [review]
Good job @anso-odoo 👍 
Just a comment below.

**Review by Brieuc-brd — APPROVED** [review]
@anso-odoo 
LGTM

**Comment by anso-odoo** [comment]
@sobo-odoo Pinged your for review because you are set as a reviewer on the task 👍  Thanks already !

File: addons/website/views/snippets/s_product_list.xml [hunk 4]
```diff
@@ -3,62 +3,95 @@
...
+    <section class="s_product_list pt64 pb64" data-vcss="001">
+        <div class="container">
             <div class="row">
+                <div class="col-lg-12">
+                    <h3>Our finest selection</h3>
+                </div>
                 <div class="col-lg-2 col-md-4 col-6">
-                    <a t-att-href="url">
-                        <img src="/web/image/website.s_product_list_default_image_1" alt="" class="img img-fluid"/>
+                    <a t-att-href="url" class="text-decoration-none">
```
Comment (sobo-odoo) [inline addons/website/views/snippets/s_product_list.xml:13]: Behavior: As soon as we click on a column, the `<a>` element disappears, but the card is still there (so when we save afterwards, the card is not clickable) :open_mouth: 

Note that were planning to implement a new column to card option (allowing to transform columns into cards) and with that there would be a stretch option, allowing to set a link on the whole card... but unfortunately, there was not enough time to implement it (yet) :grimacing: 

So in the mean time, while this option does not exist (yet), maybe set the links like before, so one on the image and one on the text ? But we will need to fix the image because the ratio does not seem to work on a `figure > a > img` :grimacing:  I will make a PR for that :+1: 

You could also use the `stretched-link` class (this is what makes a link be applied on the whole card) :arrow_down: 
```xml
<div class="col-lg-2 col-md-4 col-6 o_colored_level">                  
  <div class="s_card card o_cc o_cc1 o_colored_level o_card_img_top" data-name="Card" data-vcss="001">
      <figure class="o_card_img_wrapper ratio mb-0 ratio-4x3">
          <img src="/web/image/website.s_product_list_default_image_1" alt="" class="o_card_img card-img-top" loading="lazy" data-mimetype="image/jpeg" data-original-id="237" data-original-src="/website/static/src/img/snippets_demo/s_product_list_1.jpg" data-mimetype-before-conversion="image/jpeg" style="">
      </figure>
      <div class="card-body text-center">
          <h6 class="card-title o_default_snippet_text" style="text-align: center;">
              <a href="http://google.com" class="stretched-link text-decoration-none">Elegant</a>
          </h6>
      </div>
  </div>
</div>
```
But given that there is no way to reproduce it in the editor, I don't think you can (to discuss with @qsm-odoo maybe) :grimacing:

File: addons/website/views/snippets/s_product_list.xml [hunk 5]
```diff
@@ -3,62 +3,95 @@
...
         </div>
     </section>
 </template>
 
+<record id="website.s_product_list_001_scss" model="ir.asset">
+    <field name="name">Product list 001 SCSS</field>
+    <field name="bundle">web.assets_frontend</field>
+    <field name="path">website/static/src/snippets/s_product_list/001.scss</field>
+</record>
+
```
Comment (sobo-odoo) [inline addons/website/views/snippets/s_product_list.xml:94]: Nitpick: maybe move this just under or above the 000 SCSS asset, so they are grouped :slightly_smiling_face:

**Review by sobo-odoo — CHANGES_REQUESTED** [review]
Here is my review :slightly_smiling_face:  (sorry for the wait :grimacing:) The new design is really cool :star_struck: 
However, there are things to change, see review comments :slightly_smiling_face: 

I didn't review the design-themes PR yet, but given that the structure of the snippet might change, I will let you adapt it according to my comments here, and once ready, I will review it :wink:

**Comment by stefanorigano** [comment]
Hello @sobo-odoo , thanks so much for your insights on this, it will help other PRs as well 👍 
I'm wondering if we shouldn't tickle https://github.com/odoo/odoo/pull/176684 first, since it affects (almost) any other snippet using an `s_card` as column's only descendant  🤔
Let me know, thanks!

**Comment by sobo-odoo** [comment]
@anso-odoo @stefanorigano The ["three columns and comparisons" PR](https://github.com/odoo/odoo/pull/176684) has been merged, you can therefore rebase this one and consider my review now :wink:

**Comment by stefanorigano** [comment]
> @stefanorigano Shouldn't this snippet be in the "Columns" tab (currently in "Content") ?

TL;DR; => "Columns" it's ok for me. @lebl-odoo ?

---

The snippet showcase "Content" by using a "Columns" layout... but if pictures it's what you want to prioritise, then "Images" could fit as well ofc 🥲 ... so yeah, this is one of those snippets that highlight how much these categories could be arbitrary.
Luckily we'll re-organize everything when things will settle a bit.

**Comment by lebl-odoo** [comment]
@stefanorigano shouldn't we add it in "Product" category? Just kidding.

Go for "Columns". As you said "Luckily we'll re-organize everything when things will settle a bit.", once we'll see all snippets, we'll have a better overview and we could perform some changes afterwards.

**Comment by sobo-odoo** [comment]
@anso-odoo 
> It looks like the editor is deleting the <a> (like you described https://github.com/odoo/odoo/pull/172768#discussion_r1739297865). Do you think it is a bug or an issue in my code ? This behaviour is on master s_product_list as well.

It seems to be because the `<a>` element is not inside a `<p>` (already the case before :grimacing:) => putting it in a `<p>` should solve the issue :slightly_smiling_face:

**Comment by sobo-odoo** [comment]
@anso-odoo My bad, this is in fact because of the the `t-set` of the url => indeed, it should be `t-value` instead of `value`
```diff
- <t t-set="url" value="url or '#'"/>
+ <t t-set="url" t-value="url or '#'"/>
```
Because of that, there was in fact no `href`, since it was not set properly :grimacing: You can already fix it in your PR, I don't know how we will fix it in stable though, I will ask :slightly_smiling_face:

**Review by sobo-odoo — CHANGES_REQUESTED** [review]
Here is a last review :wink: I will review design-themes after the changes, since it surely will need to be adapted too with some of my comments :slightly_smiling_face:

File: addons/website/views/snippets/s_product_list.xml [hunk 6]
```diff
@@ -2,56 +2,83 @@
...
+        <div class="container">
             <div class="row">
-                <div class="col-lg-2 col-md-4 col-6">
-                    <a t-att-href="url">
-                        <img src="/web/image/website.s_product_list_default_image_1" alt="" class="img img-fluid"/>
-                    </a>
-                    <div class="s_product_list_item_link">
-                        <a t-att-href="url" class="btn btn-secondary">Furniture</a>
+                <div class="col-lg-12">
+                    <h3>Our finest selection</h3>
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/website/views/snippets/s_product_list.xml]: Thanks for the feedback, made all the requested changes :)

**Review by sobo-odoo — APPROVED** [review]
It's okay with me :wink: I will add the commit to "all snippets" PR :slightly_smiling_face:

**Comment by sobo-odoo** [comment]
Commit move to
- https://github.com/odoo/odoo/pull/180736
